src/Mollie/WC/Payment/Order.php

```diff
@@ -483,15 +483,24 @@ public function onWebhookCompleted( WC_Order $order, $payment, $payment_method_t
 	public function onWebhookCanceled( WC_Order $order, $payment, $payment_method_title ) {
 
 		// Get order ID in the correct way depending on WooCommerce version
-		if ( version_compare( WC_VERSION, '3.0', '<' ) ) {
-			$order_id = $order->id;
-		} else {
-			$order_id = $order->get_id();
-		}
+        $order_id = wooCommerceOrderId($order);
 
 		// Add messages to log
-		Mollie_WC_Plugin::debug( __METHOD__ . ' called for order ' . $order_id );
+		debug( __METHOD__ . ' called for order ' . $order_id );
+
+		// if the status is Completed|Refunded|Cancelled  DONT change the status to cancelled
+        $data_helper = getDataHelper();
+        $isCompleted = $data_helper->hasOrderStatus($order, 'completed');
+        $isRefunded = $data_helper->hasOrderStatus($order, 'refunded');
+        $isCancelled = $data_helper->hasOrderStatus($order, 'cancelled');
+
+        if ( $isCompleted || $isRefunded || $isCancelled){
+            debug( __METHOD__ . ' called for payment ' . $order_id . ' has final status. Nothing to be done' );
+
+            return;
+        }
```

Comment:
I would see the right coding standard here, there's that if statement which is a bit mess.

I would also change a bit the conditional by retrieving the order status and check if the status of the order is within a specific collection.

Something like:

```
// This can be moved into another method
$order = $dataHelper->getWcOrder($orderId);
$orderStatus = $dataHelper->getOrderStatus($order);
$isFinalOrderStatus = \in_array($orderStatus, ['completed', 'refunded', 'canceled'], true);

if ($isFinalOrderStatus) {
 // Perform debug
}
```

Here the use of the `invalid` is not totally appropiated, so I added some domain to make clear the list against the order status is check is a final order status.